apps/src/templates/sectionProgressV2/ExpandedProgressDataColumn.jsx

```diff
@@ -4,67 +4,24 @@ import styles from './progress-table-v2.module.scss';
 import {studentShape} from '../teacherDashboard/teacherSectionsRedux';
 import {studentLevelProgressType} from '../progress/progressTypes';
 import {connect} from 'react-redux';
-import classNames from 'classnames';
-import FontAwesome from '../FontAwesome';
 import LevelDataCell from './LevelDataCell';
-import i18n from '@cdo/locale';
+import ExpandedProgressColumnHeader from './ExpandedProgressColumnHeader.jsx';
 
 function ExpandedProgressDataColumn({
   lesson,
   levelProgressByStudent,
   sortedStudents,
   removeExpandedLesson,
 }) {
-  const header = React.useMemo(() => {
-    // If there are only 2 levels, we only show the number so that the text fits the cell.
-    const headerText =
-      lesson.levels.length < 3
-        ? lesson.relative_position
-        : i18n.lessonNumbered({
-            lessonNumber: lesson.relative_position,
-            lessonName: lesson.name,
-          });
+  const [expandedChoiceLevels, setExpandedChoiceLevels] = React.useState([]);
 
-    // Manual width is necessary so that overflow text is hidden and lesson header exactly fits levels.
-    // Add (numLevels + 1)px to account for borders.
-    const width =
-      lesson.levels.length * styles.levelCellWidth +
-      lesson.levels.length +
-      1 +
-      'px';
-    return (
-      <div className={styles.expandedHeader}>
-        <div
-          className={classNames(
-            styles.gridBox,
-            styles.expandedHeaderLessonCell
-          )}
-          style={{width}}
-          onClick={() => removeExpandedLesson(lesson.id)}
-          aria-label={headerText}
-        >
-          <FontAwesome
-            icon="caret-down"
-            className={styles.expandedHeaderCaret}
-          />
-          <div className={styles.expandedHeaderLessonText}>{headerText}</div>
-        </div>
-        <div className={styles.expandedHeaderSecondRow}>
-          {lesson.levels.map(level => (
-            <div
-              className={classNames(
-                styles.gridBox,
-                styles.expandedHeaderLevelCell
-              )}
-              key={lesson.id + '.' + level.bubbleText + '-h'}
-            >
-              {lesson.relative_position + '.' + level.bubbleText}
-            </div>
-          ))}
-        </div>
-      </div>
-    );
-  }, [removeExpandedLesson, lesson]);
+  const toggleExpandedChoiceLevel = levelId => {
+    if (expandedChoiceLevels.includes(levelId)) {
+      setExpandedChoiceLevels(expandedChoiceLevels.filter(l => l !== levelId));
+    } else {
+      setExpandedChoiceLevels([...expandedChoiceLevels, levelId]);
+    }
+  };
```

Comment:
I think this helps separate the concerns of the different components, making it so that each individual level box only has the ability to toggle its own level, and only if it is actually a choice level.

```suggestion
  const toggleExpandedChoiceLevel = level => () => {
    if (expandedChoiceLevels.includes(level.id)) {
      setExpandedChoiceLevels(expandedChoiceLevels.filter(l => l !== level.id));
    } else if (level.sublevels?.length > 0) {
      setExpandedChoiceLevels([...expandedChoiceLevels, levelId]);
    }
  };
```

Reply:
In order to switch to the `level => () => {` strategy, I would have to move this function definition from the column to each cell, I think it makes it much cleaner if it is on the column. However, I did switch to using the level instead of levelId and each cell should only know the level that it is rendering so I don't think there is any concern about modifying a different level.